src/hooks/useLoginEmailForm.ts

```diff
@@ -0,0 +1,52 @@
+// src/hooks/useLoginForm.ts
+import { useState, useCallback, type ChangeEvent, type FocusEvent, useEffect } from 'react';
+
+interface LoginIdForm {
+  id: string;
+  idError: string;
+  handleIdChange: (e: ChangeEvent<HTMLInputElement>) => void;
+  handleIdBlur: (e: FocusEvent<HTMLInputElement>) => void;
+}
+
+export const useLoginEmailForm = (): LoginIdForm => {
+  const [id, setId] = useState<string>(''); // ID 상태
+  const [idError, setIdError] = useState<string>(''); // ID 입력 오류 메시지 상태
+  const [idTouched, setIdTouched] = useState<boolean>(false); // ID input 태그 클릭 상태 -> 기본값 : 클릭되지 않은 상태
+
+  const isValidEmail = (email: string): boolean => {
+    // 정규 표현식을 통한 이메일 형식 검사
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    return emailRegex.test(email);
+  };
+
+  // ID 유효성 검사 로직
+  useEffect(() => {
+    if (!idTouched)
+      setIdError(''); //input 태그가 클릭되지 않은 상황에서 Error 값이 없으면 검사 통과
+    else if (!id.trim()) {
+      setIdError('ID를 입력해주세요.');
+    } else if (!isValidEmail(id)) {
+      setIdError('ID는 이메일 형식으로 입력해주세요.');
+    } else {
+      setIdError('');
+    }
```

Comment:
코드를 일관성 있게 작성해보시면 좋을거 같아요

```suggestion
    if (!idTouched) {
      setIdError(''); //input 태그가 클릭되지 않은 상황에서 Error 값이 없으면 검사 통과
    } else if (!id.trim()) {
      setIdError('ID를 입력해주세요.');
    } else if (!isValidEmail(id)) {
      setIdError('ID는 이메일 형식으로 입력해주세요.');
    } else {
      setIdError('');
    }
```